### ISSUE TITLE
Feature Request: User-submitted modules section

### ISSUE THREAD

**Issue by BustedSec** [trustedsec/ptf#238]
I love how easy PTF makes it to install a lot of the programs I need when setting up a new machine. Since I do this a lot, I was thinking it would be nice to have a user-submitted modules section that would allow me to automate other installation steps. For instance, I hate the new naming schema for interfaces - i.ie. en0s25 instead of eth0, so I edit my /etc/default/grub to something like GRUB_CMDLINE_LINUX="net.ifnames=0 biosdevname=0" and then run update grub. Writing a module to do that would be easy, however this is probably not a change everyone would want. Maybe I want to install conky with a custom config also, but this might not be something everyone wants. Therefore, having a section for user-submitted modules could make it easier for people to install stuff they like and customize their systems without forcing those changes on everyone who uses the framework who does an update_upgrade_all. Perhaps if someone likes a specific subset of programs and have it work similar to the way ninite does where you pick your programs with a checkbox and then it generates a custom ptf installer for those programs ... could even have a PTF script that just install their favorite programs without all the other stuff -  but thats another idea. I really love this framework.

**Comment by BustedSec** [comment]
I attempted to implement what I described above. 

As there is no easy way to edit a config file from the command line, I found a perl script that does the job pretty well and added it under the src folder. Then I wrote a installer module to change the interface names back to the old naming convention. Tested it on my current ubuntu build (16.04) as well as an older ubuntu VM (15.10) and a debian 8.6 VM, seems to work for all. Requires a reboot. See Pull request - if this isn't up to spec, let me know and I'll make whatever changes are necessary
